src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -21,35 +47,378 @@ const ActorInit En_Kakasi3_InitVars = {
     (ActorFunc)EnKakasi3_Update,
     (ActorFunc)EnKakasi3_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/EnKakasi3_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/EnKakasi3_Init.s")
+void EnKakasi3_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi3* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnKakasi3_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi3* this = THIS;
+
+    osSyncPrintf("\n\n");
+    // Translates to: Obonur -- Related to the name of the scarecrow (Bonooru)
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ おーボヌール ☆☆☆☆☆ \n" VT_RST);
+    this->actor.unk_1F = 6;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060065B0, &D_06000214, NULL, NULL, 0);
+    this->actor.flags |= 0x400;
+    this->rot = this->actor.posRot.rot;
+    this->actor.colChkInfo.mass = 0xFF;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = func_80A911F0;
+}
+
+void func_80A90E28(EnKakasi3* this) {
+    this->unk_1A4 = 0;
+    this->skelAnime.animPlaybackSpeed = 0.0f;
+    this->unk_1AA = this->unk_1AE = 0;
+
+    Math_SmoothDownscaleMaxF(&this->skelAnime.animCurrentFrame, 0.5f, 1.0f);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.x, this->rot.x, 5, 0x2710, 0);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->rot.y, 5, 0x2710, 0);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.z, this->rot.z, 5, 0x2710, 0);
+}
+
+void func_80A90EBC(EnKakasi3* this, GlobalContext* globalCtx, s32 arg) {
+    s16 currentFrame;
+    s16 phi_v0;
+
+    phi_v0 = globalCtx->msgCtx.unk_E410[0];
+    if (arg != 0) {
+        if (this->unk_19C[3] == 0) {
+            this->unk_19C[3] = (s16)Math_Rand_ZeroFloat(10.99f) + 30;
+            this->unk_1A6 = (s16)Math_Rand_ZeroFloat(4.99f);
+        }
+
+        this->unk_19A = (s16)Math_Rand_ZeroFloat(2.99f) + 5;
+        phi_v0 = this->unk_1A6;
+    }
+    switch (phi_v0) {
+        case 0:
+            this->unk_19A++;
+            if (this->unk_1A4 == 0) {
+                this->unk_1A4 = 1;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_ROLL);
+            }
+            break;
+        case 1:
+            this->unk_19A++;
+            this->unk_1B8 = 1.0f;
+            break;
+        case 2:
+            this->unk_19A++;
+            if (this->unk_1AE == 0) {
+                this->unk_1AE = 5000;
```

Comment:
`unk_1AE` and `unk_1AA` are used as rotation target values so hexadecimal should be used, even if the decimal values are nicer.

Same for parameters in calls such as `Math_SmoothScaleMaxMinS(&this->actor.shape.rot.x`, the numeric arguments should be written in hex when it's about rotations